### PR TITLE
feat(esl-media-query): normalize data instead of fail

### LINKED ISSUE #1044: [🚀esl-media-query]: normalize data instead of fail
As an `ESLMediaRuleList` consumer I want to have a little bit more smart tuple syntax parsing, to cover common cases.

**Business case:**
The component uses a shared or fixed mask to create an `ESLMediaRuleList` based value, but  the value appears to be single (equal for each case of the media query)

**Current behavior:** 
`ESLMediaRuleList` throws an error if the query and values cartages have a different size
```
ESLMediaRuleList.parse('@sm|@md', '1') // throws error
```

**Proposed behavior:**
`ESLMediaRuleList` duplicate the last value if values cortege is smaller than the queries one

**Priority**: Major

**Additional context**
- Provide related test cases
- Do not extend media-query module code base with more then a couple of lines to normalize the data

### PR DESCRIPTION
**Pull request by Natalie-Smirnova**
add condition for case if data in `valueList`  less than in `queries`

Closes: #1044

### CODE REVIEW

File: src/modules/esl-media-query/core/esl-media-rule-list.ts [hunk 1]
```diff
@@ -130,7 +130,8 @@ export class ESLMediaRuleList<T = any> extends SyntheticEventTarget {
   public static parseTuple(mask: string, values: string, parser: RulePayloadParser<any> = String): ESLMediaRuleList {
     const queries = mask.split('|');
     const valueList = values.split('|');
-    if (valueList.length !== queries.length) throw new Error('Value doesn\'t correspond to mask');
+      while (valueList.length < queries.length && valueList.length !== 0 ) valueList.push(valueList[valueList.length - 1]);
```
Comment (ala-n) [inline src/modules/esl-media-query/core/esl-media-rule-list.ts:133]:
```suggestion
      while (valueList.length < queries.length && valueList.length !== 0) valueList.push(valueList[valueList.length - 1]);
```

File: src/modules/esl-media-query/test/esl-media-rule-list.test.ts [hunk 2]
```diff
@@ -124,6 +124,46 @@ describe('ESLMediaRuleList', () => {
     mockMdMatchMedia.matches = true;
     expect(mrl.value).toBe('3');
   });
+  test('all|@sm|@md ==> 1|2|2', () => {
```
Comment (ala-n) [inline src/modules/esl-media-query/test/esl-media-rule-list.test.ts:127]:
```suggestion
  test('all|@sm|@md ==> 1|2', () => {
```

File: src/modules/esl-media-query/test/esl-media-rule-list.test.ts [hunk 3]
```diff
@@ -124,6 +124,46 @@ describe('ESLMediaRuleList', () => {
...
+
+    mockSmMatchMedia.matches = false;
+    mockMdMatchMedia.matches = true;
+    expect(mrl.value).toBe('2');
+
+    mockSmMatchMedia.matches = true;
+    mockMdMatchMedia.matches = true;
+    expect(mrl.value).toBe('2');
+  });
+  test('all|@sm|@md ==> 1|1|1', () => {
```
Comment (ala-n) [inline src/modules/esl-media-query/test/esl-media-rule-list.test.ts:147]:
```suggestion
  test('all|@sm|@md ==> 1', () => {
```

**Review by ala-n — APPROVED** [review]

**Review by dshovchko — APPROVED** [review]

File: src/modules/esl-media-query/core/esl-media-rule-list.ts [hunk 4]
```diff
@@ -130,7 +130,8 @@ export class ESLMediaRuleList<T = any> extends SyntheticEventTarget {
   public static parseTuple(mask: string, values: string, parser: RulePayloadParser<any> = String): ESLMediaRuleList {
     const queries = mask.split('|');
     const valueList = values.split('|');
-    if (valueList.length !== queries.length) throw new Error('Value doesn\'t correspond to mask');
+      while (valueList.length < queries.length && valueList.length !== 0) valueList.push(valueList[valueList.length - 1]);
```
Comment (abarmina) [inline src/modules/esl-media-query/core/esl-media-rule-list.ts:133]: cosmetic-cosmetic - extra space
```suggestion
    while (valueList.length < queries.length && valueList.length !== 0) valueList.push(valueList[valueList.length - 1]);
```

**Review by abarmina — APPROVED** [review]

**Pull request merged**